src/js/view/view.js

```diff
@@ -828,6 +832,64 @@ function View(_api, _model) {
         displayClickHandler.revertAlternateClickHandlers();
     };
 
+    const _pip = function(model, state) {
+        if (state) {
+            _this.requestPip();
+        } else if (document.pictureInPictureElement) {
+            document.exitPictureInPicture();
+        }
+    };
+
+    function removePipListeners() {
+        if (_pipVideoListeners) {
+            const { video, enter, leave, timeout } = _pipVideoListeners;
+            clearTimeout(timeout);
+            video.removeEventListener('loadedmetadata', enter);
+            video.removeEventListener('leavepictureinpicture', leave);
+        }
+    }
+
+    this.requestPip = function (videoElement) {
+        const video = videoElement || _model.get('mediaElement');
+        if (video.requestPictureInPicture) {
+            removePipListeners();
+            const enter = () => {
+                removePipListeners();
+                video.requestPictureInPicture().then(() => {
+                    if (!_pipEnabled) {
+                        _pipEnabled = true;
+                        _this.trigger('pipEnter', { video });
+                    }
+                    video.addEventListener('leavepictureinpicture', leave);
```

Comment:
I think there's a chance this event listener is not always getting removed since it's done when `removePipListeners` is called in `enter` or `requestPip`. When you leave pip mode though, it looks like `requestPip` is not called. Maybe we should also remove it when `leave` is called.